core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -135,12 +144,72 @@ private AwsSigningConfig signingConfig(SignRequest<?, ? extends AwsCredentialsId
         }
 
         if (!isPayloadSigning) {
-            signingConfig.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.UNSIGNED_PAYLOAD);
+            if (isChunkEncoding) {
```

Comment:
nit: nested if-else is a bit hard to read. Can we split up the block a bit?

```
if (isPayloadSigning) {
    configurePayloadSigning();
} else {
   configureNonPayloadSigning();
}
```

Reply:
I'm wondering if we can handle complex decisions like this in a more clean way, any ideas? I think it's easy to reason about right now, but it would be nice if we could do (pseudocode):
```
STREAMING_UNSIGNED_TRAILER = !isPayloadSigning & isChunked & isTrailing
STREAMING_SIGNED = isPayloadSigning & isChunked
...
switch (payloadType)
  case STREAMING_SIGNED:
    // configure here
  case STREAMING_UNSIGNED_TRAILER:
    // configure here
  ...
   default:
      throw new UnsupportedOperationException()
```

We could probably do a bitmask, but might not be as easy to reason through.